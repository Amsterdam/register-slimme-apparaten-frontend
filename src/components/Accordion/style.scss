$white: #ffffff;
$gray-light: #e6e6e6;
$gray-light-focus: #b7b7b7; // = 20% darker e5e5e5, missing color name in design
$whitespace-half: 7.5px;
$whitespace-base: 15px;

.c-accordion {
	margin-top: 30px;
	margin-bottom: 30px;
}

.c-accordion__content {
  display: block;
}

.c-accordion__toggle-title {
	display: inline-block;
	margin: 0;
}

.c-accordion__toggle-icon {
  width: 14px;
  height: 14px;
  vertical-align: middle;
}

.c-accordion--closed {
  .c-accordion__content {
    display: none;
  }
}

.c-accordion--dark {
  color: $white;

  .c-accordion__toggle-title {
    font-size: 20px;
  }

  .c-accordion__content {
    margin-top: $whitespace-half;
    margin-left: $whitespace-base;
  }

  .c-accordion__toggle-icon {
    margin-right: 6px;
    margin-bottom: 2px;
    transform: rotate(90deg);
    fill: $white;
  }

  &.c-accordion--closed {
    .c-accordion__toggle-icon {
      transform: rotate(-90deg);
    }
  }
}

.c-accordion--light {
  border: 2px solid $gray-light;

  &:hover,
  &:focus {
    border-color: $gray-light-focus;

    .c-accordion__toggle {
      background-color: $gray-light-focus;
    }
  }

  // Start webkit border hack: Removes white space between border and background when zooming in webkit browsers
  .c-accordion__toggle {
    &:not(:focus) {
      outline: 1px solid $gray-light;
    }
  }

  &:hover {
    .c-accordion__toggle {
      outline: 1px solid $gray-light-focus;
    };
  }
  // End webkit border hack

  .c-accordion__content {
		padding: 15px 20px;
		font-size: 18px;
		line-height: 24px;

		ul li, ol li {
			color: black;
		}
  }

  .c-accordion__toggle {
    padding: 12px 1.5 * $whitespace-half;
    background-color: $gray-light;
    font-size: 18px;
  }

  .c-accordion__toggle-icon {
    margin-top: 8px;
    float: right;
    transform: rotate(-90deg);
  }

  &.c-accordion--closed {
    .c-accordion__toggle-icon {
      transform: rotate(90deg);
    }
  }
}
