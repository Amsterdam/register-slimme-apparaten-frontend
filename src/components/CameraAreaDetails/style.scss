$primary-light: #fff;
$primary-dark: #000;
$secondary-gray80: #333;
$secondary-gray60: #666;
$secondary-gray40: #999;
$secondary-gray20: #ccc;
$secondary-gray4: #f6f6f6;
$secondary-gray5: #fafafa;

$area-details-max-width: 285px;
$area-details-top: 62px;
$area-details-right: 10px;
$area-details-bottom: 10px;

.area-details {
  background-color: $secondary-gray5;
  border: solid 1px $secondary-gray20;
  box-shadow: 2px 2px 0 0 $secondary-gray40;
  display: flex;
  flex-direction: column;
  font-family: "AvenirNextLTW01-Regular";
  max-height: calc(100% - #{$area-details-top + $area-details-bottom});
  max-width: $area-details-max-width;
  min-width: 285px;
  position: fixed;
  top: $area-details-top;
  right: $area-details-right;
  z-index: 1020;

  @media screen and (max-width:576px) {
    max-width: 100%;
    left: 0;
    top: auto;
    right: 0;
    bottom: 0;
  }

  &__heading {
    padding: 4px;
    min-height: 33px;
    display: flex;
    align-items: center;
    justify-content: flex-end;
    background-color: $primary-light;
    border-bottom: 1px solid $secondary-gray20;
  }

  &__table {
    .area-details__header-row {
      background-color: $secondary-gray60;
      box-shadow: none;
    }
  }

  &__row {
    padding: 8px;
    position: relative;
    display: flex;
    flex-direction: column;
    font-weight: normal;
  }

  &__row-label {
    padding: 0;
    font-size: 14px;
    font-weight: normal;
    line-height: 16px;
    color: $secondary-gray40;
    border: none;
    box-shadow: none;

    &:hover {
      background-color: inherit;
      box-shadow: none;
    }

    .area-details__header-row & {
      color: $primary-light;
      background-color: inherit;
    }
  }

  &__row-element {
    padding: 4px 0 0 0;
    font-size: 16px;
    line-height: 1;
    color: $secondary-gray80;
    border: none;
    box-shadow: none;

    &:hover {
      background-color: inherit;
      box-shadow: none;
    }

    .area-details__header-row & {
      font-family: 'Avenir LT W01 85 Heavy';
      color: $primary-light;
      background-color: inherit;
    }
  }

  &__question-mark-button {
    width: 24px;
    height: 24px;
    padding: 0;
    position: absolute;
    top: 24px;
    right: 5px;
    border: solid 1px $secondary-gray20;

    svg {
      width: 24px;
      height: 24px;
    }

    &:hover, &:focus {
      background-color: #ccc;
      border-color: #000 !important;
      color: #000 !important;
      outline: none;
      cursor: pointer;

      svg path {
        fill: #000;
      }
    }
  }

  &__button {
    padding: 0;
    border-radius: 0;
    height: 24px;
    margin-left: 4px;
    border: none;
    width: 24px;

    &-icon {
      fill: $secondary-gray60;
      height: inherit;
      width: inherit;

      &:hover, &:focus {
        fill: #000;
      }
    }

    &:hover, &:focus {
      background-color: #ccc;
      border-color: #000 !important;
      color: #000 !important;
      fill: #000;
    }
  }

  &__body {
    display: flex;
    flex-direction: column;
    min-height: 0; // NB: firefox fix https://stackoverflow.com/a/28639686

    // IE FIX
    @media all and (-ms-high-contrast: none), (-ms-high-contrast: active) { //sass-lint:disable-line no-vendor-prefixes
      -ms-flex: 0 0 auto; //sass-lint:disable-line no-vendor-prefixes
      -ms-flex-negative: 0; //sass-lint:disable-line no-vendor-prefixes
    }
  }

  button.action {
    padding: 8px;
    margin: 16px auto 24px auto;
    display: flex;
    align-items: center;
    font-size: 16px;
    font-weight: 900;

    svg {
      margin-right: 8px;
    }
  }
}
