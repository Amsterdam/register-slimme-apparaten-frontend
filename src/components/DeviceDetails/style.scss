$primary-light: #fff;
$primary-dark: #000;
$secondary-gray80: #333;
$secondary-gray60: #666;
$secondary-gray40: #999;
$secondary-gray20: #ccc;
$secondary-gray4: #f6f6f6;
$secondary-gray5: #fafafa;

$device-details-max-width: 285px;
$device-details-top: 62px;
$device-details-right: 10px;
$device-details-bottom: 10px;

.device-details {
  background-color: $secondary-gray5;
  border: solid 1px $secondary-gray20;
  box-shadow: 2px 2px 0 0 $secondary-gray40;
  display: flex;
  flex-direction: column;
  font-family: 'Avenir Next';
  max-height: calc(100% - #{$device-details-top + $device-details-bottom});
  max-width: $device-details-max-width;
  min-width: 285px;
  position: fixed;
  top: $device-details-top;
  right: $device-details-right;
  z-index: 1020;

  @media screen and (max-width: 576px) {
    max-width: 100%;
    left: 0;
    top: auto;
    right: 0;
    bottom: 0;
  }

  &__heading {
    padding: 4px;
    min-height: 33px;
    display: flex;
    align-items: center;
    justify-content: flex-end;
    background-color: $primary-light;
    border-bottom: 1px solid $secondary-gray20;
  }

  &__body {
    padding-bottom: 8px;
  }

  &__table {
    .device-details__header-row {
      background-color: $secondary-gray60;
      box-shadow: none;
    }
  }

  &__row {
    padding: 8px;
    position: relative;
    display: flex;
    flex-direction: column;
    font-weight: normal;
  }

  &__row-label {
    padding: 0;
    font-size: 14px;
    font-weight: normal;
    line-height: 16px;
    color: $secondary-gray40;
    border: none;
    box-shadow: none;

    &:hover {
      background-color: inherit;
      box-shadow: none;
    }

    .device-details__header-row & {
      color: $primary-light;
      background-color: inherit;
    }
  }

  &__row-element {
    padding: 4px 0 0 0;
    font-size: 16px;
    line-height: 1;
    color: $secondary-gray80;
    border: none;
    box-shadow: none;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;

    &:hover {
      background-color: inherit;
      box-shadow: none;
    }

    .device-details__header-row & {
      font-weight: 700;
      color: $primary-light;
      background-color: inherit;
    }
  }

  &__question-mark-button {
    width: 24px;
    height: 24px;
    padding: 0;
    position: absolute;
    top: 24px;
    right: 5px;
    border: solid 1px $secondary-gray20;

    svg {
      width: 24px;
      height: 24px;
    }

    &:hover,
    &:focus {
      background-color: #ccc;
      border-color: #000 !important;
      color: #000 !important;
      outline: none;
      cursor: pointer;

      svg path {
        fill: #000;
      }
    }
  }

  &__button {
    padding: 0;
    border-radius: 0;
    height: 24px;
    margin-left: 4px;
    border: none;
    width: 24px;

    &-icon {
      fill: $secondary-gray60;
      height: inherit;
      width: inherit;

      &:hover,
      &:focus {
        fill: #000;
      }
    }

    &:hover,
    &:focus {
      background-color: #ccc;
      border-color: #000 !important;
      color: #000 !important;
      fill: #000;
    }
  }

  &__body {
    display: flex;
    flex-direction: column;
    min-height: 0; // NB: firefox fix https://stackoverflow.com/a/28639686

    // IE FIX
    @media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
      -ms-flex: 0 0 auto; //sass-lint:disable-line no-vendor-prefixes
      -ms-flex-negative: 0; //sass-lint:disable-line no-vendor-prefixes
    }
  }

  button.action {
    padding: 8px;
    margin: 16px auto 24px auto;
    font-size: 16px;
    font-weight: 900;

    svg {
      margin-right: 8px;
      vertical-align: middle;
    }
  }
}
