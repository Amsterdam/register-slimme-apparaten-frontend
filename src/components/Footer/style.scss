.footer-component {
  max-width: 1023px;
  padding: 20px 20px 20px 0px;
  margin-left: 86px;

  p {
    margin-top: 25px;
    margin-bottom: 25px;
  }
}
