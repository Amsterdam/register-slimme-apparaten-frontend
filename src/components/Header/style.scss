@import '~stijl/dist/scss/ams-colorpalette';
@import '~stijl/dist/scss/ams-breakpoints';

.header-component {
  height: 50px;
  padding-left: 86px;
  display: flex;
  align-items: center;
  box-shadow: 0 2px 0 0 #999999;

  .header-title {
    color: $ams-zwart;
    text-decoration: none;
  }
}
