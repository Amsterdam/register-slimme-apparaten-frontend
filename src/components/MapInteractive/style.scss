.map-component {
  position: fixed;
  width: 100%;
  height: 100vh;
  top: 0;
  left: 0;
  z-index: 100;

  #mapdiv {
    position: fixed;
    width: 100%;
    height: 100vh;
    top: 0;
    left: 0;
  }

  .embed-search {
    max-width: 328px;

    @media screen and (max-width: 576px) {
      max-width: 100%;
    }

    .invoer {
      display: flex;
      box-shadow: 2px 2px 0 0 #999;
    }

    input[type="text"] {
      height: 32px;
      border: solid 1px #ccc;
      box-shadow: none;
      padding-top: 0;
      padding-bottom: 0;
      line-height: 32px;
    }

    &__button {
      background-color: #f6f6f6 !important;
      border: solid 1px #ccc !important;
      border-left: none;

      &:hover,
      &:focus {
        background-color: #ccc !important;
        border: solid 1px #000 !important;
        color: #000 !important;
        fill: #000;
      }

      &.primary.action:before {
        content: url(../../images/icon-search-black.svg);
        display: block;
        height: 24px;
        left: 50%;
        position: absolute;
        top: 55%;
        -webkit-transform: translate(-50%, -50%);
        transform: translate(-50%, -50%);
        width: 24px;
      }
    }

    &__auto-suggest {
      width: 100%;
    }
  }

  &__loading {
    background-color: rgba(0, 0, 0, 0.2);
    display: block;
    width: calc(100% - 30px); // padding left and right of grid
    height: 100%;
    position: absolute;
    z-index: 1001;
  }
}

.about-button {
  padding: 8px;
  position: absolute;
  top: 10px;
  right: 10px;
  box-shadow: 2px 2px 0 0 #999999;
  background-color: #f6f6f6;
  border: solid 1px #cccccc;
  cursor: pointer;
  z-index: 1010;

  font-family: "AvenirNextLTW01-Regular";
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: 1;
  letter-spacing: normal;
  color: #000000;

  &:hover,
  &:focus {
    background-color: #ccc;
    border-color: #000 !important;
    color: #000 !important;
    fill: #000;
  }

  @media screen and (max-width: 540px) {
    left: 10px;
    top: auto;
    right: auto;
    bottom: 20px;
  }
}

// Override cluster design from leaflet
.marker-cluster {
  color: white;
  background-color: white !important;
  box-shadow: 1px 1px 1px #666666;

  div {
    width: 32px !important;
    height: 32px !important;
    margin-top: 4px !important;
    margin-left: 4px !important;
    background-color: #004699 !important;
  }

  span {
    line-height: 34px !important;
  }
}

path.camera-area {
  fill: #f14600;
  fill-opacity: 0.2;
  stroke: #f14600;
  stroke-opacity: 1;
  stroke-width: 1;

  &.active-element {
    stroke-width: 3;
  }
}

.leaflet-tooltip-pane {
  path.marker-highlight {
    transform: translate(3px, -4px);
    stroke: #000;
    fill: transparent;
  }
}

@media screen and (max-width: 540px) {
  .leaflet-control-zoom {
    display: none;
  }
}

.leaflet-marker-icon.highlight {
  box-shadow: 0 0 0 3px rgba(0,0,0,0.8);
  border-radius: 50%;
}

.leaflet-control-zoom a:hover::before {
  background-image: none !important;
}

@media screen and (max-width: 576px) {
  #about-iot {
    position: fixed;
    bottom: 0;
    left: 0;
    width: 100vw;
    z-index: 1010;
  }
}
