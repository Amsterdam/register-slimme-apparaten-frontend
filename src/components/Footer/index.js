import React from 'react';

import './style.scss';

function Footer() {
  return (
    <div className="container-fluid footer-wrapper no-print">
      <div className="row bg-darkgrey footer">
        <div className="col-12 col-sm-10 offset-sm-1 col-md-8">
          <footer className="footer-component">
            <h3 className="c-footer__heading">Disclaimer</h3>
            <ul>
              <li>Het register is niet volledig. Registratie van slimme apparaten in het register is niet verplicht, wordt wel gevraagd.</li>
              <li>De gegevens over slimme apparaten in het register worden overgenomen van de eigenaar. Die is verantwoordelijk voor de juistheid en volledigheid van de gegevens.</li>
              <li>De maker van de software, OIS van de gemeente Amsterdam is niet verantwoordelijk voor de kwaliteit n het functioneren en van de inhoud van het register.</li>
              <li>De Gemeente Amsterdam, OIS kan echter niet aansprakelijk worden gesteld voor de juistheid, volledigheid en actualiteit van de website. De Gemeente Amsterdam, OIS kan in het bijzonder niet aansprakelijk worden gesteld voor eventuele schade of consequenties ontstaan door direct of indirect gebruik van de inhoud van de website.</li>
            </ul>
          </footer>
        </div>
      </div>
    </div>
  );
}

export default Footer;
