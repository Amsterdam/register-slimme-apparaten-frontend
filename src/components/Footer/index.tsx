import { Footer as AscFooter, FooterTop, Row, Column, List, ListItem, Link } from '@amsterdam/asc-ui';
import './style.scss';

const disclaimers = [
  {
    id: 1,
    value: `Het register biedt geen compleet overzicht van alle slimme apparaten. Registratie van slimme apparaten in het register is niet verplicht, maar wordt wel gevraagd.`,
  },
  { id: 2, value: `De "slimme apparaten" in het register zijn van diverse organisaties.` },
  { id: 3, value: `De gegevens van de eigenaar van het slimme apparaat worden niet getoond i.v.m. privacy.` },
  { id: 4, value: `De eigenaar bepaalt of hij reageert; hij is dat niet verplicht.` },
  {
    id: 5,
    value: `Uw mail wordt niet gearchiveerd in het register. Ook uw persoonlijke gegevens worden niet geregistreerd in het register.`,
  },
  {
    id: 6,
    value: `De gegevens van de apparaten in het register worden overgenomen van de eigenaar. Die is verantwoordelijk voor de juistheid en volledigheid van de gegevens.`,
  },
  {
    id: 7,
    value: `OIS, onderdeel van Gemeente Amsterdam, kan niet aansprakelijk worden gesteld voor de juistheid, volledigheid en actualiteit van de website. De Gemeente Amsterdam, OIS kan in het bijzonder niet aansprakelijk worden gesteld voor eventuele schade of consequenties ontstaan door direct of indirect gebruik van de inhoud van de website.`,
  },
];
const DisclaimerLinks: React.FC<{ indent?: boolean }> = () => (
  <List>
    {disclaimers.map(({ id, value }) => (
      <ListItem key={id}>
        <Link darkBackground href="/" inList>
          {value}
        </Link>
      </ListItem>
    ))}
  </List>
);
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
const Footer = () => (
  <AscFooter>
    <FooterTop>
      <Row>
        <Column wrap span={{ small: 1, medium: 2, big: 4, large: 8, xLarge: 8 }}>
          <DisclaimerLinks />
        </Column>
      </Row>
    </FooterTop>
  </AscFooter>
);

export default Footer;
