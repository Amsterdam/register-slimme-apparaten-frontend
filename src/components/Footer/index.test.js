import React from 'react';
import { shallow } from 'enzyme';

import Footer from './index';
import './style.scss';

describe('<Footer />', () => {
  it('should render correctly', () => {
    const renderedComponent = shallow(
      <Footer />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
