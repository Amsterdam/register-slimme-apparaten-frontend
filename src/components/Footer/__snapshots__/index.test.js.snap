// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render correctly 1`] = `
<div
  className="footer-wrapper no-print"
>
  <div
    className="row bg-darkgrey footer"
  >
    <footer
      className="container footer-component"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <h3
            className="c-footer__heading"
          >
            Contact
          </h3>
          <p
            className="u-mb--1"
          >
            Heeft u een vraag over de informatie die u aantreft op Mijn Amsterdam?
            <br />
            Neem dan contact op met de gemeente.
          </p>
          <p
            className="u-mb--1"
          >
            Gemeentelijk informatienummer: 
            <a
              className="c-link c-link--inherit"
              href="tel:14020"
            >
              14 020
            </a>
            <br />
            op werkdagen van 08:00 tot 18:00 uur.
          </p>
        </div>
      </div>
    </footer>
  </div>
  <div
    className="container grid-below-footer"
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <nav>
          <ul
            className="links horizontal left"
          >
            <li>
              <a
                href="https://www.amsterdam.nl/privacy/"
              >
                <span
                  className="linklabel"
                >
                  Privacy
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
